schema.yml

```diff
@@ -0,0 +1,831 @@
+          content:
+            application/json:
+              schema:
+                type: object
+  
+  /beacon-search:
+    post:
+      description: Search for datasets or persons
+      operationId: beacon_operations.post_search
+      requestBody:
+        $ref: '#/components/requestBodies/BeaconRequestBody'
+      responses:
+        200:
+          description: Response completed successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SuccessBody'
+        default:
+          $ref: '#/components/responses/BeaconErrorResponse'
+      tags:
+        - POST Endpoints
+  
+  /datasets:
+    get:
+      summary: List datasets. With parameters, searches through all datasets by one or more parameters.
+      description: |
+        Gets a list of all datasets. When a search parameter is given, this will search through all
+        datasets instead. The granularity of the response is dependent on the authorized level
+        of access of the user
+      operationId: dataset_operations.search
+      parameters:
+          - $ref: "#/components/parameters/treatmentParam"
+          - $ref: "#/components/parameters/primarySiteParam"
+          - $ref: "#/components/parameters/drugNameParam"
+          - $ref: "#/components/parameters/chrParam"
+          - $ref: "#/components/parameters/geneParam"
+          - $ref: "#/components/parameters/assemblyParam"
+          - $ref: "#/components/parameters/excludeProgramsParam"
+          - $ref: "#/components/parameters/pageSizeParam"
+          - $ref: "#/components/parameters/pageParam"
+      responses:
+        200:
+          description: Retrieved dataset information
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/DatasetAggregateBody'
+                  - $ref: '#/components/schemas/DatasetRowLevelBody'
+        400:
+          $ref: "#/components/responses/400BadRequestError"
+        401:
+          $ref: "#/components/responses/401UnauthorizedError"
+        404:
+          $ref: "#/components/responses/404NotFoundError"
+        5XX:
+          $ref: "#/components/responses/5xxServerError"
+      tags:
+        - datasets
+  /datasets/{id}:
+    get:
+      summary: Retrieve a dataset according to its ID
+      description: |
+        Gets a specific dataset by ID. Granularity of the response is
+        dependent on the authorized level of access of the user
+      operationId: dataset_operations.get_by_id
+      parameters:
+      - in: path
+        name: id
+        required: true
+        schema:
+          title: Dataset Id
+          type: string
+      responses:
+        200:
+          description: Retrieved dataset information
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/DatasetAggregateBody'
```

Comment:
A dataset might contains lots of different types of data, so I think we will want more response schemas, or more endpoints within datasets to allow users to say "only clinical data" or "clinical + variants" or "images only". Maybe something like `datasets/{id}/clinical` or `datasets/{id}/variants`?